tests/monotouch-test/CoreImage/FilterTest.cs

```diff
@@ -109,6 +117,47 @@ public void ColorSpace ()
 				Assert.Null (f.ColorSpace, "ColorSpace/reset-null");
 			}
 		}
+
+		[Test]
+		public void CIBarcodeDescriptorTest ()
+		{
+			TestRuntime.AssertXcodeVersion (9, 0);
+
+			using (var f = new CIBarcodeGenerator ()) {
+				Assert.Null (f.BarcodeDescriptor, "CIBarcodeDescriptor/default");
+				using (var b = new CIQRCodeDescriptor ()) {
+					var nsname = NSString.CreateNative ("inputBarcodeDescriptor");
+					// TODO: remove when `CIBarcodeDescriptorTest` passes
+					// This doesn't work, `inputBarcodeDescriptor` is not set.
```

Comment:
Unclear, you mean `f.BarcodeDescriptor == null` after `f.BarcodeDescriptor = b;` ?

The filter might have validation that requires other properties to be set before applying the generator ?

Reply:
@spouliot I copied the `ColorSpace` test, there's a `f.ColorSpace = cs;` followed by a `f.ColorSpace = null;` there.

I'll dig into "the filter might have validation that requires other properties to be set", didn't think about that.